### DISCUSSION TITLE
TLS handshake error from client

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#1497] [Q&A]
Hello there, I just ran a pocketbase instance on a digital ocean VPS. All seemed work. But, when I try to visit admin page, browser says: 
```
Secure Connection Failed
An error occurred during a connection to {VPS IP}. Peer reports it experienced an internal error.
Error code: SSL_ERROR_INTERNAL_ERROR_ALERT
* The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
* Please contact the website owners to inform them of this problem.
```

And pocketbase `errors.log` saays:

```
http: TLS handshake error from {CLIENT_IP}: acme/autocert: missing server name
```

Any idea how to fix this? I'm using Pocketbase v0.10.4

**Comment by ganigeorgiev** [comment]
I assume that you've started the executable with the `--https` flag. Do you have a valid public domain name or are you using the VPS IP?

Note that Let's encrypt **works only with domain names**, aka. they don't support issuing certificates on bare IPs.

**Reply by Mr-M1M3** [reply]
Seems like, this is the issue. I'm using VPS IP.

**Reply by Mr-M1M3** [reply]
@ganigeorgiev is there any workaround?

**Reply by ganigeorgiev** [reply]
Sorry, I'm not aware of workarounds for this. You'll need a domain name if you want to use the `--https` flag and the Let's encrypt certificate.

You can try looking for other free certificate authorities that don't have this limitation, but you'll have to [setup a reverse proxy](https://pocketbase.io/docs/going-to-production/#using-reverse-proxy) (nginx/caddy/apache/etc.) so that you can use them with PocketBase.

**Reply by Mr-M1M3** [reply]
@ganigeorgiev Hello again, I bought a domain and now I see different error. I asked help in another discussion (#1510) to make it easier for others to find solution for similar problems.

**Comment by jimafisk** [comment]
Similar issue:
- I'm able to view the admin dashboard in the browser fine :heavy_check_mark: 
- and valid SSL/TLS certs are being issued :heavy_check_mark: 
- however in my `pb_data/error.log` I see tons of errors :x: :

```
http: TLS handshake error from {CLIENT_IP:PORT}: acme/autocert: missing server name
```

In my systemd service I have something like:
```
ExecStart      = /path/to/pocketbase serve --http="mysub.mydomain.org:80" --https="mysub.mydomain.org:443"
```

Any idea why `acme/autocert` is so noisy? Thanks!

**Reply by ganigeorgiev** [reply]
I'm not sure what could be the cause of the error, but if it is working fine for you when you access it from the browser or sending curl requests, then I suspect that the above is a client-side error - trying to access the web server via its IP instead of the domain name, uses an old client/browser that tries to communicate with unacceptable TLS version by the server (_we currently use [the Go default](https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/crypto/tls/common.go;l=694-704), which when acting as server is min TLS 1.0; with the next release I'm [planning to bump the min version to TLS1.2](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=34884377) as some users reported that the legacy 1.0 and 1.1 support could affect the cert reputation_).

In other words, the above errors could be a result of a bot traffic or a network scanner and it should be safe to ignore them.

If it is a public resource, you can also run an online test for your certificate using something like https://www.ssllabs.com/ssltest/ to extra check the issued certificate.
